markdown.dtx

```diff
@@ -15641,6 +15765,79 @@ function M.writer.new(options)
 % \par
 % \begin{markdown}
 %
+% Define \luamdef{writer->jekyllData} as a function that will transform an
+% input YAML table `d` to the output format. The table is the value of
+% the key `p` in a parent table; if `p` is nil, then the table has no parent.
+% Any string values encountered in the table will be transformed using the
+% function `t`.
+%
+% \end{markdown}
+%  \begin{macrocode}
+  function self.jekyllData(d, t, p)
+    if not self.is_writing then return "" end
+
+    local buf = {}
+
+    local keys = {}
+    for k, _ in pairs(d) do
+      table.insert(keys, k)
+    end
+    table.sort(keys)
+
+    if not p then
+      table.insert(buf, "\\markdownRendererJekyllDataBegin")
+    end
+
+    if #d > 0 then
+        table.insert(buf, "\\markdownRendererJekyllDataSequenceBegin{")
+        table.insert(buf, p or "")
```

Comment:
We should apply some minimal escaping here, so that the users can't crash TeX outside hybrid mode by including special characters (`\`, `%`, etc.).
```suggestion
        table.insert(buf, self.uri(p or ""))
```